README
NAME
    Dist::Zilla::Plugin::Author::KENTNL::Prereqs::Latest::Selective -
    [DEPRECATED] Selectively upgrade a few modules to depend on the version
    used.

VERSION
    version 1.001001

SYNOPSIS
            [Autoprereqs]

            [Author::KENTNL::Prereqs::Latest::Selective]

    This will automatically upgrade the minimum required version to the
    currently running version, for a selective list of packages, wherever
    they appear in dependencies.

    Currently, the list of packages that will be upgraded to the current
    version are as follows:

    *   Test::More - What I test all my packages with

    *   Module::Build - The Installer I use for everything

    *   Dist::Zilla::PluginBundle::Author::KENTNL - The configuration setup
        I use for everything.

DESCRIPTION
    This module is deprecated and no longer used by @Author::KENTNL

    Instead, he recommends you use "[Prereqs::MatchInstalled]"

METHODS
  wanted_latest
            my $hash = $plugin->wanted_latest();

    A "Hash" of Modules I want to be "Latest I've released with"

            {
                    'Test::More' => 1,
                    'Module::Build' => 1,
                    'Dist::Zilla::PluginBundle::Author::KENTNL' => 1,
            }

  current_version_of
            my $v = $plugin->current_version_of('Foo');

    Returns the currently installed version of a given thing.

  for_each_dependency
            $plugin->for_each_dependency( $cpan_meta, sub {
                    my ( $self, $info ) = @_;

                    printf "%s => %s\n", $_ , $info->{$_} for qw( phase type package requirement )
            });

    Utility for iterating all dependency specifications.

    Each dependency spec is passed as a "HashRef"

            {
                    phase => 'configure',
                    type  => 'requires',
                    package => 'Module::Metadata',
                    requirement => bless({}, 'CPAN::Meta::Requirements::_Range::_Range'); # or close.
            }

  register_prereqs
    This module executes during "prereqs" generation.

    As such, its advised to place it after other things you want "prereq"'s
    upgraded on.

    ( Presently, it won't matter if you place it before, because it does
    some magic with phase emulation, but that might be removed one day )

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for Dist-Zilla-Plugin-Author-KENTNL-Prereqs-Latest-Selective

1.001002 2017-03-09T08:11:51Z 34543f6

1.001001 2017-03-09T08:10:59Z 4ac6519
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot
 - Avoid use of Test::KENTNL in tests

 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +13 ↑4 -3 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1 (recommends: +1)
 - test: +1 ↓1 -1 (recommends: ↑1 ↓1)

1.001000 2014-08-30T12:52:09Z
 [00 Minor]
 - metadata
 - tests
 - internals

 [Dependencies::Stats]
 - Dependencies changed since 1.000001, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - runtime: -1
 - test: +2

 [Internals]
 - DZPU:EmulatePhase no longer used.

 [Metadata]
 - fix author to kentnl

 [Tests]
 - Coverage testing

1.000001 2014-08-15T21:15:05Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - whitespace adjustments

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)

 [Misc]
 - Improve source side POD to reduce build side whitespace.

1.000000 2014-08-02T05:44:00Z
 [00 Major]
 - Yes. Still Deprecated.
 - Changes that affect hand-operated downstreams.

 [01 Version Scheme]
 - Migrated to x.yyyyyy
 - This is the primary reason for the 1.x on the box
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [02 Tooling]
 - Migrated to EUMM for releasing
 - Tooling dependencies softened to be 'recommended' for latest things.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.1.2, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +2 ↓1 (recommends: +3)

0.1.2 2013-11-20T03:06:09Z
 [00 Bugfix]
 - Make sure PrereqSource role is a dep

 [Dependencies::Stats]
 - Dependencies changed since 0.1.1, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - runtime: +1

0.1.1 2013-11-18T12:20:20Z
 [00 Minor]
 - Deprecate
 - Drop 5.10 minimum

 [Dependencies::Stats]
 - Dependencies changed since 0.1.0, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +50 -1 (recommends: -1, suggests: +1 -1)
 - runtime: ↓1
 - test: ↑1 -2

 [Documentation]
 - Add README.mkdn
 - Copyright 2013
 - Add MetaPOD

 [Meta]
 - Bugs to github issues

 [Packaging]
 - Use new Module::Build features

 [Tests]
 - Compile tests now ::PerFile
 - Regen CPAN-Changes test
 - Regen Kwalitee test
 - Regen ReportVersions::Tiny test

0.1.0 2012-07-07T08:44:42Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "[DEPRECATED] Selectively upgrade a few modules to depend on the version used.",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Author-KENTNL-Prereqs-Latest-Selective",
  "dist_version" => "1.000000",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Author::KENTNL::Prereqs::Latest::Selective",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Util::EmulatePhase" => 0,
    "Module::Data" => 0,
    "Moose" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.4206",
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "[DEPRECATED] Selectively upgrade a few modules to depend on the version used.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Author-KENTNL-Prereqs-Latest-Selective",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Author::KENTNL::Prereqs::Latest::Selective",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Module::Data" => 0,
    "Moose" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001002",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Role::PrereqSource" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Data" => 0,
  "Moose" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
